src/app/related-search/related-search.component.spec.ts

```diff
@@ -77,7 +77,18 @@ describe('RelatedSearchComponent', () => {
     fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it('should create an instance', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should display results if any, along with right keyword', () => {
+    let compiled = fixture.debugElement.nativeElement;
+    if (component.results.length > 0) {
```

Comment:
![image](https://user-images.githubusercontent.com/15216503/27833667-6372c330-60f1-11e7-9dc8-123582dacbea.png)

this will give default results without initialising, i.e we have results=[] initialized to empty array. due to which the tests won't run at all.
actually, we should avoid conditional statements while writing tests.